docs/DeveloperGuide.md

```diff
@@ -2,37 +2,731 @@
+
+Given below is an example usage scenario and how StorageHandler behaves at each step:
+
+1. The user launches the application for the first time. Group is initialized, creating an instance of StorageHandler. StorageHandler creates relevant storage directories if they do not yet exist.
+2. StorageHandler reads data from the 2 data storage files and creates Member and Transaction objects in the associated utility list objects.
+3. When a command which would alter the data within MemberList or TransactionList is invoked, the method to save data to the storage files is called by Group. This updates the information within the storage files.
+
+<ins>Design Considerations</ins>
+
+* Update upon change, not upon exit - This allows for data to be saved even if the application exits ungracefully.
+* *checkTransactions* - Methods are provided to have a quick check to ensure that data from data storage is not corrupted.
+
+### Member and MemberList
+
+### Transaction and TransactionList
+<ins>Transaction Overview</ins>
+
+The Transaction class is responsible for representing a single transaction in the LongAh application between 2 members.
+It contains information about the lender, borrowers, and the amount involved in the transaction.
+
+
+<ins>Class Fields</ins>
+* lender: Represents the member who lent the money.
+* transaction time (optional): Represents the time at which the transaction took place.
+* subtransactions: An ArrayList of Subtransaction objects, representing individual borrowings within the transaction.
+
+<ins>Transaction Constructor</ins>
+
+`Transaction(String userInput, MemberList members)`
+* Parses the given user input and creates a new Transaction object with the specified lender, borrowers, and amount.
+* Called whenever a new transaction is added to the transaction list.
+
+`Transaction(Member lender, ArrayList<Subtransaction> subtransactions,
+MemberList members)`
+* Constructs a transaction instance using specified lender, subtransactions, member list.
+* This constructor is used for storage methods only.
+
+`Transaction(Member lender, ArrayList<Subtransaction> subtransactions,
+MemberList members, String transactionTime)`
+* Constructs a transaction instance using specified lender, subtransactions, member list and transaction time.
+* This constructor is used for storage methods only on entries with transaction time.
+
+<ins>Transaction Methods</ins>
+
+- *parseTransaction*: Parses the user input to extract lender and borrowers, then adds them to the transaction.
+
+- *addBorrower*: Adds a borrower to the transaction.
+
+- *getLender*: Returns the lender of the transaction.
+
+- *isLender*: Checks if a given member is the lender of the transaction.
+
+- *isborrower*: Checks if a given member is a borrower in the transaction.
+
+- *isInvolved*: Checks if a given member is involved in the transaction.
+
+- *toStorageString*: Converts the transaction to a string format for storage.
```

Comment:
I think in contrast to code that we write, documents could use more indentations, so that it's easy to spot headers. For example, you could indent the lines with <ins> tags.